internal/util/nf_authorization.go

```diff
@@ -0,0 +1,22 @@
+package util
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/free5gc/openapi/oauth"
+	udr_context "github.com/free5gc/udr/internal/context"
+)
+
+func AuthorizationCheck(c *gin.Context, serviceName string) error {
```

Comment:
move it to context.go:
```
func (context *UDRContext) AuthorizationCheck(authHdr string) error {
	if !context.OAuth2Required {
		return nil
	}
	err := oauth.VerifyOAuth(authHdr, "nudr-dr", context.NrfCertPem)
	if err != nil {
		return err
	}
	return nil
}
```

Reply:
I would keep the ```serviceName``` as a parameter. 
In case the other NFs have more than one service and have the same logic/code as UDR.

Reply:
```go
func (context *UDRContext) AuthorizationCheck(authHdr string, serviceName string) error {
	if !context.OAuth2Required {
		return nil
	}
	err := oauth.VerifyOAuth(authHdr, serviceName, context.NrfCertPem)
	if err != nil {
		return err
	}
	return nil
}
```